src/QMCDrivers/DMC/WalkerControl.cpp

```diff
@@ -236,13 +208,15 @@ void WalkerControl::branch(int iter, MCPopulation& pop, bool do_not_branch)
     throw std::runtime_error("Potential bug! Population num_global_walkers mismatched!");
 #endif
 
+  // This defensive code implies that previous code left population walkers in invalid state.
   if (!do_not_branch)
     for (UPtr<MCPWalker>& walker : pop.get_walkers())
     {
       walker->Weight       = 1.0;
       walker->Multiplicity = 1.0;
     }
 
+  // This is debugging code, should be removed.
```

Comment:
This not debugging code. This was used to avoid unnecessary recompute.

Reply:
Ah makes sense... I will reread

Reply:
Left some documentiation behind so this can be returned to in the ParticleSet refactoring.